tower/src/util/boxed/mod.rs

```diff
@@ -32,8 +32,9 @@
 //! [`Service`]: crate::Service
 //! [`Rc`]: std::rc::Rc
 
+mod layer;
 mod sync;
 mod unsync;
 
 #[allow(unreachable_pub)] // https://github.com/rust-lang/rust/issues/57411
-pub use self::{sync::BoxService, unsync::UnsyncBoxService};
+pub use self::{layer::BoxLayer, sync::BoxService, unsync::UnsyncBoxService};
```

Comment:
do we think there's much of a use-case for an `UnsyncBoxLayer` as well? I think what we _might_ want is a `Layer` that is `Sync` (and still `Arc`ed), but that produces `UnsyncBoxService`s instead of `BoxService`s. 

but, i guess that's just something we can easily add later if it turns out anyone needs it...